gulpfile.cjs

```diff
@@ -1724,6 +1728,140 @@ function createSpecList() {
   fs.writeFileSync(path.join("Specs", "SpecList.js"), contents);
 }
 
+/**
+ * Reads `ThirdParty.extra.json` file
+ * @param path {string} Path to `ThirdParty.extra.json`
+ * @param discoveredDependencies {Array<string>} List of previously discovered modules
+ * @returns {Promise<Array<Object>>} A promise to an array of objects with 'name`, `license`, and `url` strings
+ */
+function getLicenseDataFromThirdPartyExtra(path, discoveredDependencies) {
+  if (!fs.existsSync(path)) {
+    return Promise.resolve([]);
```

Comment:
Shouldn't we fail here and put out a message since the `ThirdParty.extra.json` should always exist?